### PR TITLE
Added onlyUserCanView

### PR DESCRIPTION
**Pull request by standup75**
I thought onlyAdminCanView should call this new onlyUserCanView instead of onlyRoleCanView, but maybe I am missing something?

I would also be happy to add unit test, but the suite is timing on 21 tests: `Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.`

    slouch
      attachment
        1) "before each" hook for "should create attachment from base 64 data"
        2) "after each" hook for "should create attachment from base 64 data"
      config
        ✓ should set for couchdb 1
        ✓ should set for couchdb 2
        ✓ should set for couchdb 2 when multiple nodes
        ✓ should request for max nodes
        3) should couch_httpd_auth/timeout
        4) should get and unset
        5) should unset ignore missing
        6) should set couch_httpd_auth/allow_persistent_cookies
        7) should set compaction rule
        8) should set max_dbs_open
        9) should set httpd/max_connections
      db
        10) "before each" hook for "should check if exists"
      doc
        11) "before each" hook for "should create, update and get doc"
        12) "after each" hook for "should create, update and get doc"
      enhanced-request
        ✓ should get 404 status code
        ✓ should handle malformed error
        ✓ should reconnect when all DBs active
        ✓ should throw error reach max retries (167ms)
        ✓ should ignore default_authentication_handler errors when requesting
        ✓ should set max connections
        ✓ should handle ENOTFOUND errors (43ms)
        ✓ should handle ECONNREFUSED errors
      exclude-design-docs-iterator
        13) "before each" hook for "should filter"
        14) "after each" hook for "should filter"
      membership
        15) should get membership in couchdb 1
      request-wrapper
        ✓ should not replace existing headers
        ✓ should set cookie in header
      security
        16) "before each" hook for "should set and get security"
        17) "after each" hook for "should set and get security"
      system
        18) "before each" hook for "should clone params when falsy"
        19) "after each" hook for "should clone params when falsy"
      user
        20) "before each" hook for "should not create when already exists"
        21) "after each" hook for "should not create when already exists"

### CODE REVIEW

**Comment by redgeoff** [comment]
This looks good on the surface, but I don't think you should be changing `onlyAdminCanView` as it refers to the admin role. It is possible that this is causing issues with the test suite.

We'll definitely need a test as we need to make sure to maintain 100% coverage.

**Comment by standup75** [comment]
Ok, I add tests, and documented some issues I went through running the tests (I used a fresh install of couchdb). I still have 1 issue running the tests, related to http://127.0.0.1:5984/_db_updates returning

    {
      "error": "not_found",
      "reason": "Database does not exist."
    }

Is _db_updates a module set needs setup or something?

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+### How to run the tests?
```
Comment (redgeoff) [inline test/README.md]: I think the majority of these details are already in TESTING.md. I've just added a link to TESTING.md on the README.

**Comment by redgeoff** [comment]
Looking good. Yeah, the _db_updates depends on _global_changes.

Also, as per the code review comments, https://github.com/redgeoff/slouch/blob/master/TESTING.md has some of the testing details that you we're probably missing. I've added a link to the README. TESTING.md includes how to create the system DBs via the [run-couchdb-docker.sh script](https://github.com/redgeoff/slouch/blob/master/run-couchdb-docker.sh#L22)

**Comment by redgeoff** [comment]
And I just [added some more to TESTING.md](https://github.com/redgeoff/slouch/blob/master/TESTING.md#option-a-automated-setup-via-vagrant) about setting up a dev env. I'm a big fan of using vagrant. Slouch is still quite new so the docs still need to be filled in :)

**Comment by standup75** [comment]
Make sense, yes, totally miss that file... Removed test/README.md

**Comment by redgeoff** [comment]
Looks like the assert-beautified test is failing because the new/changed code hasn't been beautified: https://github.com/redgeoff/slouch/blob/master/TESTING.md#beautify

**Pull request merged**